src/BoundaryConditions/field_boundary_conditions.jl

```diff
@@ -127,14 +124,13 @@ and the topology in the boundary-normal direction is used:
  - `ImpenetrableBoundaryCondition` for `Bounded` directions and `Face`-located fields
  - `nothing` for `Flat` directions and/or `Nothing`-located fields
 """
-FieldBoundaryConditions(default_bounded_bc::BoundaryCondition = NoFluxBoundaryCondition();
-                        west = DefaultBoundaryCondition(default_bounded_bc),
-                        east = DefaultBoundaryCondition(default_bounded_bc),
-                        south = DefaultBoundaryCondition(default_bounded_bc),
-                        north = DefaultBoundaryCondition(default_bounded_bc),
-                        bottom = DefaultBoundaryCondition(default_bounded_bc),
-                        top = DefaultBoundaryCondition(default_bounded_bc),
-                        immersed = DefaultBoundaryCondition(default_bounded_bc)) =
+FieldBoundaryConditions(; west = DefaultBoundaryCondition(),
+                          east = DefaultBoundaryCondition(),
+                          south = DefaultBoundaryCondition(),
+                          north = DefaultBoundaryCondition(),
+                          bottom = DefaultBoundaryCondition(),
+                          top = DefaultBoundaryCondition(),
+                          immersed = DefaultBoundaryCondition()) =
```

Comment:
I'd like to go back to the other design. It is meant to provide more generality for model developers, even though it does not apply to Oceananigans now. Some models may want to use `nothing` as the default.